lib/knapsack_pro/adapters/rspec_adapter.rb

```diff
@@ -54,71 +56,66 @@ def self.order_option(cli_args)
         parsed_options(cli_args)&.[](:order)
       end
 
-      def self.test_path(example)
-        example_group = example.metadata[:example_group]
-
-        if defined?(::Turnip) && Gem::Version.new(::Turnip::VERSION) < Gem::Version.new('2.0.0')
-          unless example_group[:turnip]
-            until example_group[:parent_example_group].nil?
-              example_group = example_group[:parent_example_group]
-            end
+      def self.file_path_for(example)
+        [
+          # https://github.com/rspec/rspec-core/blob/1eeadce5aa7137ead054783c31ff35cbfe9d07cc/lib/rspec/core/example.rb#L122
+          -> { example.id.match(/\A(.*?)(?:\[([\d\s:,]+)\])?\z/).captures.first },
+          -> { example.metadata[:file_path] },
+          -> { example.metadata[:example_group][:file_path] },
+          -> { top_level_group(example)[:file_path] },
+        ]
+          .each do |path|
+            p = path.call
+            return p if p.include?('_spec.rb')
           end
-        else
-          until example_group[:parent_example_group].nil?
-            example_group = example_group[:parent_example_group]
-          end
-        end
 
-        example_group[:file_path]
+        return ''
+      end
+
+      # private
+      def self.top_level_group(example)
+        group = example.metadata[:example_group]
+        until group[:parent_example_group].nil?
+          group = group[:parent_example_group]
+        end
+        group
       end
 
       def bind_time_tracker
-        ::RSpec.configure do |config|
-          config.prepend_before(:context) do
-            KnapsackPro.tracker.start_timer
-          end
+        ensure_no_focus!
+        log_subset_duration
+      end
 
+      def ensure_no_focus!
+        ::RSpec.configure do |config|
           config.around(:each) do |example|
-            current_test_path = KnapsackPro::Adapters::RSpecAdapter.test_path(example)
-
-            # Stop timer to update time for a previously run test example.
-            # This way we count time spent in runtime for the previous test example after around(:each) is already done.
-            # Only do that if we're in the same test file. Otherwise, `before(:all)` execution time in the current file
-            # will be applied to the previously ran test file.
-            if KnapsackPro.tracker.current_test_path&.start_with?(KnapsackPro::TestFileCleaner.clean(current_test_path))
-              KnapsackPro.tracker.stop_timer
-            end
-
-            KnapsackPro.tracker.current_test_path =
-              if KnapsackPro::Config::Env.rspec_split_by_test_examples? && KnapsackPro::Adapters::RSpecAdapter.slow_test_file?(RSpecAdapter, current_test_path)
-                example.id
```

Comment:
Regarding the [issue](https://github.com/KnapsackPro/knapsack_pro-ruby/pull/229#issuecomment-1830474240).

The above code was tracking the test case path (like `a_spec.rb[1:2]`).

Reply:
Fix: https://github.com/KnapsackPro/knapsack_pro-ruby/pull/229#discussion_r1408290857